octobot_commons/tree/event_tree.py

```diff
@@ -0,0 +1,207 @@
+# pylint: disable=R1725,W0221
+#  Drakkar-Software OctoBot-Commons
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import asyncio
+
+import octobot_commons.tree.base_tree as base_tree
+import octobot_commons.logging as logging
+
+
+class EventTreeNode(base_tree.BaseTreeNode):
+    """
+    Node element of an EventTreeNode. self.node_value is an asyncio.Event() that is triggered when all of its children
+    are triggered or is manually triggered. Adding an unset child will clear self. Children updates will overwrite
+    any manual trigger
+    """
+
+    __slots__ = [
+        "_parent",
+        "_logger",
+    ]
+
+    def __init__(self, *_, triggered=False, **__):
+        super().__init__(asyncio.Event(), asyncio.Event)
+        if triggered:
+            self.node_value.set()
```

Comment:
Why not using self.trigger?

Reply:
To avoid logging the trigger as it is initialized as triggered

Reply:
I think we should find an other way to prevent this log because it's no a good idea to duplicate the trigger implementation (node_value.set()) in my opinion